include/quda_openqcd_interface.h

```diff
@@ -0,0 +1,525 @@
+#pragma once
+
+#include "mpi.h"
+#include "pthread.h"
+
+#define OPENQCD_MAX_INVERTERS 32
+#define OPENQCD_MAX_EIGENSOLVERS 32
+
+/**
+ * The macro battle below is to trick quda.h to think that double_complex is
+ * defined to be the struct below. For this we need to set the __CUDACC_RTC__,
+ * which makes double_complex to be defined as double2 (see quda.h), which we
+ * redefine below as openqcd_complex_dble. The original definitions of
+ * __CUDACC_RTC__ and double2 are recovered below. We do this to be able to
+ * include this header file into a openQxD program and compile with flags
+ * "-std=C89 -pedantic -Werror". Else the compiler trows an
+ * "ISO C90 does not support complex types" error because of the
+ * "double _Complex" data types exposed in quda.h.
+ */
+
+typedef struct {
+  double re, im;
+} openqcd_complex_dble;
+
+#ifdef __CUDACC_RTC__
+#define __CUDACC_RTC_ORIGINAL__ __CUDACC_RTC__
+#endif
+
+#ifdef double2
+#define double2_ORIGINAL double2
+#endif
+
+#define __CUDACC_RTC__
+#define double2 openqcd_complex_dble
+#include <quda.h>
+#undef double2
+#undef __CUDACC_RTC__
+
+#ifdef double2_ORIGINAL
+#define double2 double2_ORIGINAL
+#undef double2_ORIGINAL
+#endif
+
+#ifdef __CUDACC_RTC_ORIGINAL__
+#define __CUDACC_RTC__ __CUDACC_RTC_ORIGINAL__
+#undef __CUDACC_RTC_ORIGINAL__
+#endif
+
+/**
+ * @file    quda_openqcd_interface.h
+ *
+ * @section Description
+ *
+ * The header file defines the interface to enable easy
+ * interfacing between QUDA and the OpenQCD software.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * OpenQCD inmitations of structs.
+ * Copied from flags.h, su3.h
```

Comment:
We can't have code directly copied from `OpenQCD`. However, I see that the data layouts are publicly documented in the `docs` subdirectory of the OpenQCD code, as is noted in the `README` file, which I didn't need to look at the code to see.

Can you remove this explicit copy+paste and implement data structures, such as `openQCD_bc_params_t`, that match the data layout in the documentation, and cite the documentation appropriately?